### PR TITLE
Synths can now charge from cyborg chargers

### PR DESCRIPTION
**Pull request by Majkl-J**
## About The Pull Request

Part 3 of my synthcentric update series (Yes we are far from done)

My insanity grows. You can probably deduce what this does, it is literally the title.

## Why It's Good For The Game

Robot charge in robot charger, good, yes?

## Changelog

:cl:
add: Synths can now charge in cyborg chargers
/:cl:

### CODE REVIEW

**Comment by nevimer** [comment]
they always should have been able to..

**Comment by Majkl-J** [comment]
> they always should have been able to..

Well they apparently cannot. The signal is nowhere to be found in their code, search for it yourself if you do not believe me.

image

File: modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	add_synth_signals(receiver)
+
+/obj/item/organ/internal/stomach/synth/Remove(mob/living/carbon/stomach_owner, special)
+	. = ..()
+	remove_synth_signals(stomach_owner)
+
+/obj/item/organ/internal/stomach/synth/proc/add_synth_signals(mob/living/carbon/stomach_owner)
+	SIGNAL_HANDLER
+
+	RegisterSignal(stomach_owner, COMSIG_PROCESS_BORGCHARGER_OCCUPANT, PROC_REF(on_borg_charge))
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm:12]: white space makes me sad :(
```suggestion
	RegisterSignal(stomach_owner, COMSIG_PROCESS_BORGCHARGER_OCCUPANT, PROC_REF(on_borg_charge))
```

File: modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+/obj/item/organ/internal/stomach/synth/proc/remove_synth_signals(mob/living/carbon/stomach_owner)
+	SIGNAL_HANDLER
+
+	UnregisterSignal(stomach_owner, COMSIG_PROCESS_BORGCHARGER_OCCUPANT)
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm:17]:
```suggestion
	SIGNAL_HANDLER
	UnregisterSignal(stomach_owner, COMSIG_PROCESS_BORGCHARGER_OCCUPANT)
```

File: modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+/obj/item/organ/internal/stomach/synth/proc/on_borg_charge(datum/source, amount)
+	SIGNAL_HANDLER
+
+	// This is so it isn't completely instant
+	amount /= 200
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/synths/code/bodyparts/stomach.dm:23]:
```suggestion
	SIGNAL_HANDLER
	amount /= 200 // This is so it isn't completely instant
```

**Review by StrangeWeirdKitten — COMMENTED** [review]
I know it's so incredibly minor, but we don't need all this extra space.

**Pull request merged**